tests/test_translate.py

```diff
@@ -112,6 +111,17 @@ def sample_wherefail(
     return "\n".join(return_strings)
 
 
+def process_override(threshold_overrides, testobj, test_name, backend):
```

Comment:
How would you feel about `platform` being optional? Something like,
```python
        plat = platform()
        if max_error is not None:
            if isinstance(max_error, dict) and plat in max_error:
                testobj.max_error = float(max_error[backend][plat])
            else:
                testobj.max_error = float(max_error[backend])
        if near_zero is not None:
            if isinstance(near_zero, dict) and plat in near_zero:
                testobj.near_zero = float(near_zero[backend][plat])
            else:
                testobj.near_zero = float(near_zero[backend])
```
Or moving it to the top of the hierarchy?
```python
    plat = platform()
    if plat in threshold_overrides:
        overrides = threshold_overrides[plat]
    else:
        overrides = threshold_overrides
    override = threshold_overrides.get(test_name, None)
    if override is not None:
        max_error = override.get("max_error", None)
        near_zero = override.get("near_zero", None)
        if max_error is not None:
            testobj.max_error = float(max_error[backend])
        if near_zero is not None:
            testobj.near_zero = float(near_zero[backend])
```
This way it will not need to be typed so often in the yaml file...

Reply:
I like that!

Reply:
alternatively we could use lists/dicts :
C_SW:
  max_error:
        - backend: numpy
          platform: docker
          value: 2e-10
      - backend: gtcuda
         value: 1e-10
XPPM:
      - platform: metal
        value: 1e-9 
   
The second array item would mean apply this to both platforms for gtcuda, the third would mean on the metal platform use this threshold with all backends for XPPM. 
What do you think of this?

Reply:
or actually, how about make the list at the test level:
C_SW:
 - max_error: 2e-10
   backend: numpy
  platform: docker 
- backend: gtcuda
  max_error: 1e-10
  near_zero: 1e-14
XPPM:
- platform: metal
   max_error: 1e-9

Reply:
Even better!